test/unit/math/gpu/cholesky_decompose_test.cpp

```diff
@@ -26,10 +26,8 @@ TEST(MathMatrix, cholesky_decompose_cpu_vs_gpu_small) {
   stan::math::matrix_d m0_res = stan::math::cholesky_decompose(m0);
   stan::math::matrix_d m1_res = stan::math::cholesky_decompose(m1);
 
-  stan::math::matrix_gpu m0_chol_gpu = stan::math::internal::cholesky_decompose(
-      m0_gpu, floor(m0.rows() * 0.04));
-  stan::math::matrix_gpu m1_chol_gpu = stan::math::internal::cholesky_decompose(
-      m1_gpu, floor(m0.rows() * 0.04));
+  stan::math::matrix_gpu m0_chol_gpu = stan::math::cholesky_decompose(m0_gpu);
```

Comment:
Would still be good to have tests for multiple combinations of multiple values of the tuning opts.

Reply:
I'll add a little test that changes the values of the tuning opts

Reply:
thanks :D

Reply:
@SteveBronder let me know if I can help with any of the requested changes

Reply:
Sure! These all should be pretty reasonable. I get off of work in like an hour or so and can slam these out tonight. 
--------- below not about this PR
Rok if you have time tonight it would be rad to finish up rev, think all we need to do is do the same tests that currently happen in rev but make the size bigger so that the GPU stuff goes off for all the tests.

Also want to look at why we currently need that if statement for `C.size() > 0` in the rev code before we make that PR. Feels like there should be an easy way to get rid of that and it has a weird smell

Reply:
Sean check out the new tests, I iterate over different tuning parameters. It takes a bit so not sure if it's too much testing and we should test like two parameters each

Reply:
Also Rok if you want to work on rev tmrw let me know else I'll wrap it up tmrw mid-afternoon

Reply:
I am going to work on it right now, its 8 AM here. If you have time just check my changes tmrw mid-afternoon your time :)